Src/WitsmlExplorer.Frontend/services/credentialsService.ts

```diff
@@ -78,76 +83,69 @@ class CredentialsService {
     return this.credentials.find((c) => c.server.id === this.sourceServer?.id);
   }
 
-  public clearPasswords() {
-    this.credentials = this.credentials.map((creds) => {
-      if (this.hasValidCookieForServer(creds.server.url)) {
-        return creds;
-      }
-      return { server: creds.server };
-    });
-    if (!this.hasValidCookieForServer(this.server.url)) {
-      this._onCredentialStateChanged.dispatch({ server: this.server, hasPassword: this.isAuthorizedForServer(this.server) });
-    }
-  }
-
   public isAuthorizedForServer(server: Server): boolean {
     if (msalEnabled && server?.securityscheme == SecurityScheme.OAuth2 && getUserAppRoles().some((x) => server.roles.includes(x))) {
       return true;
     }
-    return this.credentials.find((c) => c.server.id === server?.id)?.password !== undefined;
-  }
-
-  public hasPasswordForUrl(serverUrl: string) {
-    return this.credentials.find((c) => c.server.url === serverUrl)?.password !== undefined;
+    return this.hasValidCookieForServer(server?.url);
   }
 
   public hasValidCookieForServer(serverUrl: string): boolean {
     //use local storage to check whether the cookie is valid, because the cookie is httpOnly
-    const cookieTimestamp = localStorage.getItem(serverUrl);
-    if (!cookieTimestamp) {
+    const cookieInfo = localStorage.getItem(serverUrl);
+    if (!cookieInfo) {
       return false;
+    } else {
+      const cInfo: CookieInfo = JSON.parse(cookieInfo);
+      return new Date().getTime() < new Date(cInfo.expiry).getTime();
     }
-    return new Date().getTime() < new Date(cookieTimestamp).getTime();
-  }
-
-  public keepLoggedInToServer(serverUrl: string): boolean {
```

Comment:
Should this functionality be reimplemented instead of removed? Could save the keepLoggedIn value in CookieInfo if so.

Reply:
yes maybe we could keep it by examining the Persistent" data in localstorage, good idea

Reply:
Implemented as described above